scripts/rnnlm/train_rnnlm.sh

```diff
@@ -182,18 +185,21 @@ while [ $x -lt $num_iters ]; do
         else
           sparse_opt=''; embedding_type=word
         fi
-        if $use_gpu; then gpu_opt="--use-gpu=yes"; queue_gpu_opt="--gpu 1";
-        else gpu_opt="--use-gpu=no"; queue_gpu_opt=""; fi
+        if $use_gpu; then gpu_opt="--use-gpu=yes"; queue_gpu_opt_="$queue_gpu_opt";
+        else gpu_opt="--use-gpu=no"; queue_gpu_opt_=""; fi
         if [ $this_num_jobs -gt 1 ]; then dest_number=$[x+1].$n
         else dest_number=$[x+1]; fi
         # in the normal case $repeated data will be just one copy.
         repeated_data=$(for n in $(seq $num_repeats); do echo -n $dir/text/$split.txt ''; done)
-        
+
         rnnlm_l2_factor=$(perl -e "print (1.0/$this_num_jobs);")
         embedding_l2_regularize=$(perl -e "print ($embedding_l2/$this_num_jobs);")
 
+        # allocate queue-slots for threads doing sampling,
+        [ -f $dir/sampling.lm ] && queue_thread_opt="--num-threads $num_egs_threads" || queue_thread_opt=
```

Comment:
Normally it won't actually use that much CPU on average because it's limited by GPU time.  I don't know how many it uses though.
The num_egs_threads is more like an upper bound to make sure it is not limited by the sampling.

Reply:
Aha, at our servers, it was running with 600% CPU, while the num-threads was set to 10. So it was 'eating' the CPU time of other slots (GPU process is supposed to consum 100% CPU only).